packages/notify-client/src/controllers/engine.ts

```diff
@@ -336,6 +337,71 @@ export class NotifyEngine extends INotifyEngine {
     return true;
   };
 
+  public getNotificationHistory: INotifyEngine["getNotificationHistory"] =
+    async ({ topic, limit, startingAfter }) => {
+      this.isInitialized();
+
+      const subscription = this.client.subscriptions.get(topic);
+
+      const identityKey = encodeEd25519Key(
+        await this.client.identityKeys.getIdentity({
+          account: subscription.account,
+        })
+      );
+
+      const issuedAt = Math.round(Date.now() / 1000);
+      const expiry =
+        issuedAt + ENGINE_RPC_OPTS["wc_notifyGetNotifications"].res.ttl;
+
+      const cachedKey = this.getCachedDappKey(subscription);
+      const dappUrl = getDappUrl(subscription.metadata.appDomain);
+      const { dappIdentityKey } = cachedKey
+        ? { dappIdentityKey: cachedKey }
+        : await this.resolveKeys(dappUrl);
+
+      const getNotificationsClaims: NotifyClientTypes.GetNotificationsJwtClaims =
+        {
+          act: "notify_get_notifications",
+          aft: startingAfter ?? null,
+          iss: identityKey,
+          ksu: this.client.keyserverUrl,
+          sub: composeDidPkh(subscription.account),
+          iat: issuedAt,
+          exp: expiry,
+          aud: encodeEd25519Key(dappIdentityKey),
+          app: `${DID_WEB_PREFIX}${subscription.metadata.appDomain}`,
+          lmt: limit ?? 50,
+          // TODO: adapt to unread capabilities when available on Notify server
+          urf: false,
+        };
+
+      const auth = await this.client.identityKeys.generateIdAuth(
+        subscription.account,
+        getNotificationsClaims
+      );
+
+      return new Promise((resolve, reject) => {
+        this.once("notify_get_notifications_response", (args) => {
+          if (args.error === null) {
+            resolve(args);
+          } else {
+            reject(new Error(args.error));
+          }
+        });
+
+        // Add timeout to prevent memory leaks with undying promises
+        setTimeout(() => {
+          reject(
+            new Error("getNotificationHistory timed out waiting for a response")
+          );
+          // Using five minutes as it is the TTL of wc_getNotificationHistory
+          // The FIVE_MINUTES const is in seconds, not ms.
+        }, FIVE_MINUTES * 1000);
+
+        this.sendRequest(topic, "wc_notifyGetNotifications", { auth });
```

Comment:
All other `this.sendRequest` calls are awaited, not sure if intentional here:

```suggestion
        await this.sendRequest(topic, "wc_notifyGetNotifications", { auth });
```

Reply:
It is in a promise, can't use `await` there